test/unit/middleware/generator-middleware.test.js

```diff
@@ -187,4 +185,37 @@ describe('Generator Middleware', function () {
     await repo.history.wait()
   })
 
+  it('can push new actions during history.wait() while processing a generator', async function () {
```

Comment:
Eh, they shouldn't. Actually, I wrote this test while I was tracking down a bug on a project. Totally unrelated. This was purely for my own confirmation.

We could remove this.